packages/leva/src/components/ValueInput/ValueInput.tsx

```diff
@@ -26,6 +26,11 @@ export function ValueInput({ innerLabel, value, onUpdate, onChange, onKeyDown, t
     []
   )
 
+  const _onChangeEnd = React.useRef<null | (() => void)>(null)
+  React.useEffect(() => {
+    _onChangeEnd.current = () => onChangeEnd?.(panelValue)
+  }, [onChangeEnd, panelValue])
+
```

Comment:
I did not know whether having onChangeEnd and panelValue as dependencies of the effect below could have some unwanted implications, so I access it via a ref...